feat(cowfi): add CMS pages for all legal content

fairlighteth opened the pull request:
# Summary

Refactor legal pages from hardcoded content to generic CMS integration system. The initial goal being to update the legal page content, which now is handled through the CMS.

Migrated all legal pages (Terms, Privacy Policy, Cookie Policy, Widget Terms, Integrator Terms) from hardcoded React components to a generic CMS-driven architecture. Reduces code while enabling dynamic content management.

  # To Test

  1. **Legal Pages Navigation** - Visit `/legal`
     - [ ] Index page displays all legal page links
     - [ ] Breadcrumb navigation works: "Home > Legal > [Page Name]"
     - [ ] All links lead to properly formatted pages

  2. **Individual Legal Pages** - Test each page:
     - [ ] `/legal/cowswap-terms` - Terms content displays correctly
     - [ ] `/legal/cowswap-privacy-policy` - Privacy policy renders properly
     - [ ] `/legal/cowswap-cookie-policy` - Cookie policy shows formatted content
     - [ ] `/legal/widget-terms` - Widget terms (old page) redirects to `/legal/integrator-terms`
     - [ ] `/legal/integrator-terms` - Integrator terms displays

  3. **Content Formatting** - Verify on any legal page:
     - [ ] Markdown content renders as HTML
     - [ ] Tables display properly (not raw markdown)
     - [ ] Page titles and meta descriptions are correct
     - [ ] Content sections flow properly

  4. **Error Handling** - Check build process:
     - [ ] Pages fail to build if CMS content missing (no silent fallbacks)
     - [ ] TypeScript compilation passes without errors

  # Background

Created a fully generic CMS content system using `createCmsPage()` utility that reduces each page to 8 lines of code. The architecture scales beyond legal pages to any content type.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * CMS-driven legal pages with a reusable CMS page factory, server-rendered CMS pages, and 12‑hour content revalidation; new integrator-terms page.
  * New CMS rendering UI: markdown renderer, content page layout with breadcrumbs/analytics, styled lists and tables.

* **Refactor**
  * Replaced multiple static legal pages and minimal layouts with CMS-based rendering; updated redirects to the integrator-terms route.

* **Chores**
  * Upgraded react-markdown and remark-gfm.

* **Style**
  * Improved heading spacing and nested list indentation in CMS content.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

fairlighteth commented:
Preview URL to check https://cowfi-git-feat-cowfi-contentpages-cowswap.vercel.app/legal

alfetopito reviewed (approved):
Question: shouldn't these links open in the same page?

<img width="583" height="472" alt="image" src="https://github.com/user-attachments/assets/c558a0d9-a035-442d-a997-9ef820775028" />

Overall, looking nice 👍

fairlighteth commented:
@elena-zh thanks for reviewing.

1. This is a content issue and should be handled in the CMS by the content owner. I did change those text links to be actual URLs now.
2. Fixed.
3. See 1)
4. This is a content issue, as in Strapi CMS there's no such option. The only option is to manually add the letters in front. Also see 1)
5.  See 1)
6. See 1)
7. See 1)
8. Out of scope for now otherwise see 1)
9. Fixed (horizontal scroll like rest of the cow.fi content has for tables)
